nixos/modules/config/malloc.nix

```diff
@@ -97,7 +97,6 @@ in
   };
 
   config = mkIf (cfg.provider != "libc") {
-    boot.kernel.sysctl."vm.max_map_count" = mkIf (cfg.provider == "graphene-hardened") (mkDefault 1048576);
```

Comment:
unclear to me if you removed this because you needed for reconcilation

Reply:
If after this PR `vm.max_map_count` is already set to 1048576 by default, there is no point to a "in certain conditions, set `vm.max_map_count` 1048576".

Reply:
There is a point: the one where we drop or change the other setting and the person who does it forget about this file :).

Reply:
If this commit is reverted, then this line will return.

Reply:
This is not the point: this value could change without having this commit being reverted.

Reply:
This is a defense in depth measure against errors, as the reconciliation would happen with no extra cost except evaluation time.

Reply:
In case the value is changed on `sysctl.nix`, wouldn't two different definitions of the same option cause an evaluation error? (Since they have the same priority, `mkDefault`)

Reply:
No because integers are **reconciled** for equality of values to the best of my knowledge, but also, this is why I started "did you need to remove this for reconcilation?" :p

Reply:
I'm tired of dealing with unresolved thread which end up with a PR being merged, I will let the author of those lines decide @enc0urage.